sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -537,7 +537,7 @@ public PipelineResponse(string requestId, HttpResponseMessage responseMessage, S
             public override void Dispose()
             {
                 _responseMessage?.Dispose();
-                DisposeContentStreamIfNotBuffered();
+                DisposeStreamIfNotBuffered(_contentStream);
```

Comment:
Can you add `_contentStream = null;`

Reply:
That won't be correct. We need to leave the stream assigned.

Reply:
Once you have disposed the stream, shouldn't leave it as not null.